code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -366,3 +366,25 @@ Contains most of the procs that are called when a mob is attacked by something
 	if(!istype(C)) return
 	if(!(unique_access in C.access)) return
 	return 1
+
+
+/mob/living/carbon/human/screech_act(mob/living/carbon/Xenomorph/Queen/Q)
+	shake_camera(src, 3 SECONDS, 1) //50 deciseconds, SORRY 5 seconds was way too long. 3 seconds now
+	var/dist = get_dist(src, Q)
+
+	if(dist > 7)
```

Comment:
Clamp would also be a possible solution here

Reply:
Also, admins procalling may want a higher dist than 7, this is meant more of a sanity check for values like 100 or 1000 or 1000000000

Reply:
Hmm, good point. Just remove the check, really.